tools/CAT/src/app/features/subCategories/subCategoriesExtraReducers.ts

```diff
@@ -0,0 +1,110 @@
+import {
+	addSubCategory,
+	updateSubCategory,
+	deleteSubCategory,
+} from './subCategoriesThunks';
+import { CategoryDto } from '@/models/api';
+import { SubCategoryResState } from '@/models/subCategories';
+
+const extraReducersConfigSubCategories = (builder) => {
+	//add
+	builder.addCase(addSubCategory.pending, (state) => {
+		state.loading = true;
+		state.error = null;
+	});
+	builder.addCase(addSubCategory.fulfilled, (state, action) => {
+		const { categoryId } = action.payload;
+
+		const updatedCategories = state.categories?.map(
+			(category: CategoryDto) => {
+				if (category.id === categoryId) {
+					return {
+						...category,
+						subCategories: [
+							...category.subCategories,
+							action.payload,
+						],
+					};
+				} else {
+					return category;
+				}
+			}
+		);
+
+		return {
+			...state,
+			categories: updatedCategories,
+			loading: false,
+			error: null,
+		};
+	});
+
+	builder.addCase(addSubCategory.rejected, (state, action) => {
+		state.loading = false;
+		state.error = (action.payload as string) || 'Something went wrong';
+	});
+	//update
+	builder.addCase(updateSubCategory.pending, (state) => {
+		state.loading = true;
+		state.error = null;
+	});
+
+	builder.addCase(updateSubCategory.fulfilled, (state, action) => {
+		const { id } = action.payload;
+
+		const updatedCategories = state.categories?.map(
+			(category: CategoryDto) => {
+				const updatedSubCategories = category.subCategories.map(
```

Comment:
if (category.subCategories[id]) category.subCategories[id] = action.payload;
return category;

Reply:
This option is not suitable because I update by index and not by id, it is important for me to keep the location in the array

Reply:
gotch you my bad, please use 'find method' then